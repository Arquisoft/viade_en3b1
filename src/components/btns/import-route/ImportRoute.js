import React, { Component } from 'react';
import { Button, withStyles } from '@material-ui/core';
import { DropzoneDialog } from 'material-ui-dropzone';
import { ImportRouteForm } from '../../containers/importrouteform/ImportRouteForm';
import {Redirect, Link, withRouter} from 'react-router-dom';

export class ImportRoute extends Component {

    constructor(props) {
        super(props);
        this.state = {
            open: false,
        };
    }

    handleClose() {
        this.setState({
            open: false
        });
    }

    handleOpen() {
        this.setState({
            open: true,
        });
    }

    handleSave(files) {
        this.setState({
            open: false,
        });
        this.props.history.push({
            pathname: '/import-route',
            routes: files
          })
    }

    render() {

        const { classes } = this.props;

        return (
            <div>
                <Button color="primary" variant='contained' className={classes.btn} onClick={() => this.handleOpen()}>
                    Import Route
                </Button>
                <DropzoneDialog
                    open={this.state.open}
                    acceptedFiles={['.gpx', '.json', '.jsonld']}
                    showPreviews={true}
                    showPreviewsInDropzone={false}
                    maxFileSize={5000000}
                    filesLimit={10}
                    clearOnUnmount={true}
                    onSave={(files) => this.handleSave(files)}
                    onClose={this.handleClose.bind(this)}
                    submitButtonText={"import"}
                    cancelButtonText={"cancel"}
                    dialogTitle={"Import routes"}
                    dropzoneText={"Drag and drop any GPX or JSON route files"}
                />
            </div>
        );
    }
}

const useStyles = (theme) => ({
    btn: {
        marginRight: theme.spacing(2),
    },
});

export default withRouter(withStyles(useStyles)(ImportRoute));
