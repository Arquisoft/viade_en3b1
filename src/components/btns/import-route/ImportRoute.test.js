import ImportRoute from './ImportRoute';
import { shallow, mount } from 'enzyme';
import { getByTestId, render, fireEvent } from '@testing-library/react';

test('Define ImportRoute', () => {
    expect(ImportRoute).toBeDefined();
});

test('Renders correctly', () => {
    const wrapper = shallow(<ImportRoute.WrappedComponent />);
    expect(wrapper).toMatchSnapshot();
});

test('Methods', () => {
    const wrapper = shallow(<ImportRoute.WrappedComponent history={[]} />);
    const instance = wrapper.shallow().instance();

    instance.handleOpen();
    instance.handleClose();
    instance.handleSave();
});

test("Click import btn", () => {
    const { container } = render(<ImportRoute.WrappedComponent />);

    const button = getByTestId(container, 'btn-import-route-test');
    fireEvent.click(button);
});