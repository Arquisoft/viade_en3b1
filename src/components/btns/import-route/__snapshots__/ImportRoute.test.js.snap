// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<ImportRoute
  classes={
    Object {
      "btn": "ImportRoute-btn-1",
    }
  }
/>
`;
